tuf/utils/x509.go

```diff
@@ -252,9 +252,31 @@ func ParsePEMPublicKey(pubKeyBytes []byte) (data.PublicKey, error) {
 			return nil, fmt.Errorf("invalid certificate: %v", err)
 		}
 		return CertToKey(cert), nil
+	case "PUBLIC KEY":
+		keyType, err := keyTypeForPublicKey(pemBlock.Bytes)
```

Comment:
Would it make sense to move this entire case block into a new constructor for `data.PublicKey` objects that takes a `[]byte` (in this case `pemBlock.Bytes`) and returns `(PublicKey, error)`?